### PR TITLE
Make avgpool2d output shape consistent

### LINKED ISSUE #88144: AvgPool2D output shapes are inconsistent when ceil_mode=True
### 🐛 Describe the bug

Certain inputs produce unexpected output shapes from [AvgPool2D](https://pytorch.org/docs/stable/generated/torch.nn.AvgPool2d.html) when `ceil_mode=True`.

## Example 1

```python
>>> t = torch.ones((1, 3, 32, 32))
>>> pool = torch.nn.AvgPool2d(8, stride=4, ceil_mode=True)
>>> pool(t).shape
torch.Size([1, 3, 7, 7])
```

This is surprising behavior because there are 8 possible windows with `stride=4` (not 7) for both height and width, each starting from the following indexes:
```
0
4
8
12
16
20
24
28
```

The final window at index 28 is a partial window, but should be included when `ceil_mode=True`, otherwise `ceil_mode` seems to serve no purpose. (This window does not start in a padding region as there is no padding used here.)

## Example 2
```
>>> t = torch.ones((1, 3, 4, 4))
>>> pool = torch.nn.AvgPool2d(3, stride=3, ceil_mode=True)
>>> pool(t).shape
torch.Size([1, 3, 2, 2])
```

With `stride=3`, we have 2 potential start positions:
```
0
3
```

Torch chooses to use both of them. But based on the behavior observed in Example 1 (where the last frame was discarded) we would have expected output shape of `[1, 3, 1, 1]`.

### Versions

PyTorch version: 1.12.1+cu102
Is debug build: False
CUDA used to build PyTorch: 10.2
ROCM used to build PyTorch: N/A

OS: Amazon Linux 2 (x86_64)
GCC version: (GCC) 7.3.1 20180712 (Red Hat 7.3.1-15)
Clang version: Could not collect
CMake version: version 3.18.4
Libc version: glibc-2.2.5

Python version: 3.7.10 (default, Jun  3 2021, 00:02:01)  [GCC 7.3.1 20180712 (Red Hat 7.3.1-13)] (64-bit runtime)
Python platform: Linux-5.4.209-129.367.amzn2int.x86_64-x86_64-with-glibc2.2.5
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] numpy==1.20.0
[pip3] torch==1.12.1
[pip3] torch-neuron==1.12.1.2.0.0.0
[conda] Could not collect

### PR DESCRIPTION
**Pull request by Guobing-Chen**
Fixes #88144

The root-cause is that the original output shape calculation formula is mixed-up. Fundamentally, dilation is not relevant when in ceiling mode ---- no matter dilation param is 1 or more, the starting point of each pooling window is not impacted. The updated formula for ceiling mode only includes input_size, padding and stride, and then do the ceiling.

cc @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @chunyuan-w @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

### CODE REVIEW

**Review by mingfeima — CHANGES_REQUESTED** [review]
please also update test cases to reflect the failures reported by https://github.com/pytorch/pytorch/issues/88144 and https://github.com/pytorch/pytorch/issues/88464
